crates/core/tedge_api/src/alarm.rs

```diff
@@ -1,58 +1,84 @@
-use std::convert::TryFrom;
-use std::convert::TryInto;
-use std::fmt;
-
+use crate::entity_store::EntityMetadata;
 use clock::Timestamp;
 use serde::Deserialize;
 use serde::Serialize;
-
 use serde_json::Value;
 use std::collections::HashMap;
-/// In-memory representation of ThinEdge JSON alarm.
-#[derive(Debug, Serialize, Deserialize, Eq, PartialEq)]
+use std::convert::TryFrom;
+
+/// Internal representation of ThinEdge alarm model.
+#[derive(Debug, Eq, PartialEq)]
 pub struct ThinEdgeAlarm {
-    pub name: String,
-    pub severity: AlarmSeverity,
-    #[serde(flatten)]
-    pub data: Option<ThinEdgeAlarmData>,
-    pub source: Option<String>,
-}
+    /// Alarm type retrieved from the MQTT topic or payload.
+    /// The "type" given from the payload has priority to the one from the topic.
+    pub alarm_type: Option<String>,
```

Comment:
Why not a `String` as the empty string is a valid alarm type (for thin-edge).

Reply:
Actually it should be `String`. I was overthinking about the case like `te/device/main///a/` (empty type), but this must be `"".to_string()`, not `None`. In short, I'll fix it.

Reply:
I'm not sure if I understood @albinsuresh here correctly. So what I wanted to say here are
1. `Option<String>` is wrong. It should be `String` as @didier-wenzek suggested.
2. The value of `alarm_type` is always delivered by [Channel](https://github.com/thin-edge/thin-edge.io/blob/c0ded23e1927a156f3f83941eda8fc854c6c997f/crates/core/tedge_api/src/mqtt_topics.rs#L323). Since our MQTT scheme allows an empty string as an alarm type (at least this `Channel` implementation returns `String("")`), we need to remember this `alarm_type` may have `String("")`. 
3. If we don't want to allow `String("")`, the implementation should be done in the `Channel` level.

Reply:
Done with ec55b7328b99294efd1e2e16b7430bb7bed5a371